mockbroker.go

```diff
@@ -31,14 +31,21 @@ type MockBroker struct {
 	brokerID     int32
 	port         int32
 	stopper      chan bool
-	expectations chan encoder
+	expectations chan *BrokerExpectation
 	listener     net.Listener
 	t            TestState
-	latency      time.Duration
 }
 
-func (b *MockBroker) SetLatency(latency time.Duration) {
-	b.latency = latency
+type MockCluster map[int32]*MockBroker
```

Comment:
To make sure the index matches the BrokerID. Broker IDs are 1-indexed, while slice indices start at 0.